task3/task3_solution.py

```diff
@@ -0,0 +1,150 @@
+class Journal:
+
+    def __init__(self, students):
+        self.student_data = students
+
+    def to_letters(self, grades):
+        res = ""
+        for grade in grades:
+            if grade >= 95.0:
+                res += "A"
+            elif grade >= 85.0:
+                res += "B"
+            elif grade >= 75.0:
+                res += "C"
+            elif grade >= 65.0:
+                res += "D"
+            elif grade >= 60.0:
+                res += "E"
+            elif grade < 60.0:
+                res += "F"
+            res += " "
+        return res.strip()
+
+    # calc overall average grade
+    def avg_grade(self, grades):
+        res = sum(grades)/len(grades)
+        return res
+
+    def inDict(self, name):
+        found = False
+        for dict in self.student_data:
+            if name in dict.values():
+                found = True
+                return found
+                break
+        return found
+
+    def get_letter_grades(self, *students, throw_exception=False):
+        for name in students:
+            # if name is present in dictionaries, do stuff
+            if self.inDict(name):
+                for dict in self.student_data:
+                    if dict.get("name") == name:
+                        print("Grades of " + name + ": ")
+                        temp = dict.copy()
+                        temp.pop("name")
+                        for work, grades in temp.items():
+                            print(work, self.to_letters(grades))
+
+                        # obsolete code, for testing purposes
+                        #
+                        # print("Homework:", self.to_letters(dict["homework"]))
+                        # print("Quizzes:", self.to_letters(dict["quizzes"]))
+                        # print("Tests:", self.to_letters(dict["tests"]))
+            else:
+                # if keyword specified, show error instead of skipping
+                if throw_exception:
+                    print("No entry", name, "in table")
+
+    def get_student_average(self, *students, throw_exception=False):
+        for name in students:
+            # if name is present in dictionaries, do stuff
+            if self.inDict(name):
+                for dict in self.student_data:
+                    if dict.get("name") == name:
+                        temp = dict.copy()
+                        temp.pop("name")
+                        # there might be an overkill here or an easier way
+                        # to pass list of values to avg_grade
+                        list = []
+                        for values in temp.values():
+                            list.extend(values)
+                        print("Average grade of", name, ":",
+                              "{0:5.2f}".format(self.avg_grade(list)))

```

Comment:
1) avoid using format, try f-string
2) if you always use {.2f} for avg_grade, try moving it to method itself